lib/pbench/test/unit/server/test_relay.py

```diff
@@ -0,0 +1,318 @@
+from http import HTTPStatus
+from logging import Logger
+from pathlib import Path
+
+from flask import Request
+import pytest
+import responses
+
+from pbench.server import OperationCode, PbenchServerConfig
+from pbench.server.api.resources import ApiParams
+from pbench.server.api.resources.intake_base import Intake
+from pbench.server.api.resources.relay import Relay
+from pbench.server.cache_manager import CacheManager
+from pbench.server.database.models.audit import (
+    Audit,
+    AuditReason,
+    AuditStatus,
+    AuditType,
+)
+from pbench.server.database.models.datasets import Dataset
+from pbench.test.unit.server import DRB_USER_ID
+
+
+class TestRelay:
+    """Test the Relay API.
+
+    This focuses on testing the unique aspects of the _prepare and _access
+    methods rather than repeating coverage of all the common base class code.
+
+    In particular, failure of either of the two external GET operations to the
+    relay, and problems in the Relay configuration file.
+    """
```

Comment:
These tests seem way more "functional" than they need to be to test just `pbench.server.api.resources.relay`.  That module has only four functions in it (one of which is a simple c'tor and another is a trivial pass-through).  `_prepare()` and `_access()` do invoke `requests.get()`, so we might want to use `responses` to mock that, but our requirements are simple enough there that we could roll our own mocks for it.  In any case, we don't need to come in through `client.post()`, and so we don't need the `FakeCacheManager`, etc.

So, if we picked a more focused unit, we might be able to have simpler tests and less code here.